service/resolver.go

```diff
@@ -119,40 +119,59 @@ func (r *resolve) MapK8sResourceAthenzResource(k8sRes string) string {
 	return athenzRes
 }
 
-// createAthenzDomain use cfg.ServiceAthenzDomain;
+// createAthenzDomain use cfg.ServiceAthenzDomains;
 
+// do the following for each cfg.ServiceAthenzDomains
 // split it with ".";
 // for each token, if it match /^_.*_$/ but not "_namespace_", replace the token with config.GetActualValue(token);
 // and then return the processed value
-func (r *resolve) createAthenzDomain() string {
-	reps := make([]string, 0, strings.Count(r.cfg.ServiceAthenzDomain, ".")+1)
-	for _, v := range strings.Split(r.cfg.ServiceAthenzDomain, ".") {
-		if v != "_namespace_" && strings.HasPrefix(v, "_") && strings.HasSuffix(v, "_") {
-			// Note: If deploying in a different namespace than the kube-public namespace, change it to get information from kube api
-			reps = append(reps, v, config.GetActualValue(v))
+func (r *resolve) createAthenzDomain() []string {
+	domains := make([]string, 0, len(r.cfg.ServiceAthenzDomains))
+	for _, domain := range r.cfg.ServiceAthenzDomains {
+		reps := make([]string, 0, strings.Count(domain, ".")+1)
+		for _, v := range strings.Split(domain, ".") {
+			if v != "_namespace_" && strings.HasPrefix(v, "_") && strings.HasSuffix(v, "_") {
+				// Note: If deploying in a different namespace than the kube-public namespace, change it to get information from kube api
+				reps = append(reps, v, config.GetActualValue(v))
+			}
 		}
+		domains = append(domains, strings.NewReplacer(reps...).Replace(domain))
 	}
-	return strings.NewReplacer(reps...).Replace(r.cfg.ServiceAthenzDomain)
+	return domains
+}
+
+// BuildDomainFromNamespace return domains by processing athenzDomains.
+func (r *resolve) BuildDomainFromNamespace(namespace string) []string {
+	return r.buildAthenzDomain(r.athenzDomains, namespace)
 }
 
-// BuildDomainFromNamespace return domain by processing athenzDomain.
-// if namespace != "", replace `/ = .`, then `.. => -`, then replace "_namespace_" in athenzDomain with namespace;
-// else replace "._namespace_" in athenzDomain with namespace;
+//  BuildServiceAccountPrefixFromNamespace return domains by processing AthenzServiceACcountPrefix.
+func (r *resolve) BuildServiceAccountPrefixFromNamespace(namespace string) []string {
```

Comment:
I need the same logic as BuildDomainFromNamespace, so I added a function that creates a prefix.